csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/FilePathFilter.cs

```diff
@@ -74,16 +74,18 @@ public IEnumerable<FileInfo> Filter(IEnumerable<FileInfo> files)
                         includeByDefault)
                 });
 
-            // Move included pathfilters to the front of the list:
-            pathFilters.Sort((pf1, pf2) => -1 * pf1.Include.CompareTo(pf2.Include));
             return unfilteredResult.Where(f =>
             {
                 var include = f.FileInclusion.Include;
-                foreach (var pathFilter in pathFilters)
+                // LGTM_INDEX_FILTERS is a prioritized list, where later filters takes
+                // priority over earlier ones.
+                for (int i = pathFilters.Count - 1; i >= 0; i--)
                 {
+                    var pathFilter = pathFilters[i];
```

Comment:
Minor: I wonder if `pathFilters.Reverse()` or `foreach (var pathFilter in pathFilters.Reverse<PathFilter>())` would be clearer, rather than switching to indices.

Reply:
I actually looked on stackoverflow for this, which said that using the inline `.Reverse` in the `foreach` was quite inefficient.

Reply:
I am not sure off the top of my head if having the `.Reverse` in the `foreach` would get evaluated with every iteration, but replacing `pathFilters.Sort` with `pathFilters.Reverse` should at least be more efficient than `.Sort` (although likely not as efficient as using the indices of course).